tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -2354,7 +2386,11 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
         value match {
           case js.Function(args, body) =>
             // Make sure the function has a meaningful `name` property
-            js.FunctionDef(globalVarIdent, args, body)
+            val functionExpr = js.FunctionDef(globalVarIdent, args, body)
+            if (keepFunctionExpression)
+              js.VarDef(globalVarIdent, Some(functionExpr))
```

Comment:
I do not understand why this is necessary. I understand that this prevents hoisting of the function (probably, although the function keeps its name....). What does that change? It's not like the function body is evaluated...